### PR TITLE
French conjugator group1 & 2 present

### PR DESCRIPTION
**Pull request by cecilebertin**
Redo of French conjugator with tests.

This part will work on all the verbs of group1 (verbs ending in -er), être, avoir and aller in the present tense.

For now, the combineToConjugate() function covers all the cases of combinations between the stems of the verbs in -er and all the possible tense endings for that group.
This means combineToConjugate() already works for all the possible tenses of group1 and can be used as it is, when adding tenses to the French conjugator.
The functions findSubject(), findWordGroup(), combineToConjugate() can be used in the future for the other tenses and groups.

### CODE REVIEW

**Review by yjlim5 — APPROVED** [review]

File: src/french.js [hunk 1]
```diff
@@ -1,68 +1,434 @@
...
+      }
+  */
   conjugate (word, info) {
-    // Format for info: {tense: 'present', formal: 'true/false', gender: 'female/male/undefined', singular: 'true/false/undefined', person: '1'/'2'/'3'}
-    //TODO define 3 verb group types
+    switch (info.mood) {
+      case 'Indicatif':
+        switch (info.tense) {
+          case 'Présent':
+            return this.doIndicatifPresent(word, info);
```
Comment (songz) [inline src/french.js:43]: don't put break after return statements.

http://stackoverflow.com/questions/6330114/do-you-need-break-in-switch-when-return-is-used

File: src/french.js [hunk 2]
```diff
@@ -1,68 +1,434 @@
...
+            break;
+          case 'Passé Composé':
+            break;
+          case 'Imparfait':
```
Comment (songz) [inline src/french.js]: to use a switch statement correctly,

```
case 'Passé Composé':
...
case 'Futur Antérieur':
break;
```

Comment (songz) [reply src/french.js]: Apply to the rest of the switch statements below.

File: src/french.js [hunk 3]
```diff
@@ -1,68 +1,434 @@
...
-   switch (info.tense) {
-     case 'present':
-       return this.doPresent(word, info);
-       break;
-     default:
-       return 'Could not find any rules for conjugation';
-   }
+  findSubject (info) {
+    // returns the subject for conjugation: 's1'/'s2'/'s3'/'p1'/'p2'/'p3'
+    if (info.singular === true) {
```
Comment (songz) [inline src/french.js]: if statements execute if true. Unnecessary to compare it with truthy statement. Apply for all the code below.

```
if (info.singular)
```

File: src/french.js [hunk 4]
```diff
@@ -1,68 +1,434 @@
...
+  }
 
-   switch (info.tense) {
-     case 'present':
-       return this.doPresent(word, info);
-       break;
-     default:
-       return 'Could not find any rules for conjugation';
-   }
+  findSubject (info) {
```
Comment (songz) [inline src/french.js]: this is retrieving the subject. The standard naming convention for retrieval is get. I would call this `getSubject`

File: src/french.js [hunk 5]
```diff
@@ -1,68 +1,434 @@
...
+    // returns the subject for conjugation: 's1'/'s2'/'s3'/'p1'/'p2'/'p3'
+    if (info.singular === true) {
+      if (info.person === '2' && info.formal === true){
+        //if the subject ('you') is meant as formal and singular, it is conjugated as plural, but the past participle stays singular
+        return 'p2';
+      }
+      return 's' + info.person;
```
Comment (songz) [inline src/french.js]: return `s${info.person}`;

Comment (songz) [reply src/french.js]: return `\`s${info.person}\``

Comment (cecilebertin) [reply src/french.js]: I changed this to `s${info.person}`and changed some others too.
There were some cases where I wasn't sure if I should use a template literal or If I should just leave the expression as is.
like : return [stem.slice(0, -3) + 'èvr' + ending];
When do you know if you should use a template literal or avoid it?

**Review by songz — APPROVED** [review]
WOW! GOOD WORK!

**Comment by cecilebertin** [comment]
Thank you @songz !

**Pull request merged**